src/domaine-model.md

```diff
@@ -0,0 +1,69 @@
+| | | | | only a name was added | returns object with only a name and empty strings | `Student {studentID: 1, firstName: Steve, lastName: '', this.gitHub: '', this.email: ''}`
+| | | | | No parameter added| throws an error |`"Error: please enter a name to add student"`
+| | | findStudent(studentID(@Int))| a teacher wantes to find specific student from allStudents list |returns student object |if valid ID is inputted |`Student {studentID: 1, firstName: Steve, lastName: Stevenson, this.gitHub: Steveyboy, this.email: steve@hotmail.com}`
+| | | | |if ID is not found | throws an error |`"Error: 'student not found'`
+| | | | |if no parameter was inputted |throws an error |`"Error: 'please enter a student ID'`
+| | | addStudentToCohort(studentID(@int), cohort(@string)) | a teacher wants to specify which cohort the student belongs in|a valid studentID and cohort inputed | pushes student to corresponding cohort |`Student {studentID: 1, firstName: Steve, lastName: Stevenson, this.gitHub: Steveyboy, this.email: steve@hotmail.com}`
+| | | | |if one of the parameters is left empty |throws an error |`"Error: please enter both student and cohort`
+| | | | |if parameter does not match a student| throws an error| `'Error: "student not found"'`
+| | | | |if parameter does not match a cohort| throws an error| `'Error: "No cohort found with that name"'`
+| | | removeStudentFromCohort(studentID(@Int), cohort(@string))|a teacher would like to remove a student from a specific cohort | if a valid studentID and a cohort is inputed |The corresponding student is removed from the corresponding cohort | 'removed student no longer exists in student list'
+| | | | |if only a studentID is inputed | throws Error | `'Error: "please enter a cohort name"'`
+| | | | |if only a cohort is entered |throws an error | `"Error: please enter a student ID`
+| | | | | if student does not exist | throws an error |`'"Error: "student not found"'`
+| | | | | if cohort does not exist | throws an error |`'Error: "No cohort found with that name"'`
+| | | removeStudent(studentId(@Int))|a teacher wantes to remove student entirely | a valid id is entered|removes student from enitre cohort manager | `Student is permanently removed`
+| | | | | if student Id does not match| throws an error | `'Error: "student not found"'`
+| | | | | if no parameter is inputted | throws an error | `'Error: "please enter a student ID"'`
+
+## COHORT
+
+Objects | Properties | Methods | Notes | Scenario | Output | Example
+------- | ---------- | -------- | ----- | -------- | ------ | -------
+| | | | | | |
+| Cohort | this.name, this.studentList |||||
+| | | | | | |
+| | | | | | |
+
+## STUDENT
+
+Objects | Properties | Methods | Notes | Scenario | Output | Example
+------- | ---------- | -------- | ----- | -------- | ------ | -------
+| | | | | | |
+| Student |this.studentID, this.firstName, this.lastName, this.gitHub, this.email, this.cohorts | | | | |
```

Comment:
Detailed Domain model. Well written.

Reply:
I still think its messy and hard to update. Going to try something a bit cleaner in the future.